src/cursor/abstract_cursor.ts

```diff
@@ -184,6 +186,7 @@ export abstract class AbstractCursor<
           : ReadPreference.primary,
       ...pluckBSONSerializeOptions(options)
     };
+    this[kOptions].timeoutMS = options.timeoutMS;
```

Comment:
How are we planning on determining whether or not CSOT is enabled?  If we're testing for the presence of `timeoutMS` in the options, then we need to be careful not to set `timeoutMS` by accident when CSOT isn't enabled.

Reply:
The `timeoutMS` option on both the `AbstractCursorOptions` and the `InternalAbstractCursorOptions` being assigned to here are of type `number | undefined`, so we'll use nullishness to determine whether or not CSOT is enabled. 

Note also that nowhere in this PR is `timeoutMS` given a default non-nullish value, so unless it is explicitly provided, signaling the user's intent to enable CSOT, it will remain `undefined` and so retain legacy behaviour.